candump.c

```diff
@@ -214,6 +233,100 @@ int idx2dindex(int ifidx, int socket) {
 	return i;
 }
 
+int sprint_auto_filename_format(char * buffer)
+{
+	time_t currtime;
+	struct tm now;
+
+	if (time(&currtime) == (time_t)-1) {
+		perror("time");
+		//running = 0;
+		return 1;
+	}
+
+	localtime_r(&currtime, &now);
+
+	sprintf(buffer, "candump-%04d-%02d-%02d_%02d%02d%02d.log",
+		now.tm_year + 1900,
+		now.tm_mon + 1,
+		now.tm_mday,
+		now.tm_hour,
+		now.tm_min,
+		now.tm_sec);
+
+	fprintf(stderr, "Enabling Logfile '%s'\n", buffer);
+
+	return 0;
+}
+
+//opens file using global filehandle logfile
+int open_log_file(const char * fileName)
+{
+	if (silent != SILENT_ON)
+		fprintf(stderr, "Warning: Console output active while logging!\n");
+
+	const char * fopen_opts = (sighup_count > 0) ? "a" : "w";
+	logfile = fopen(fileName, fopen_opts);
+
+	if (!logfile) {
+		perror("logfile");
+		return 1;
+	}
+
+	return 0;
+}
+
+//flush,close then reopen log
+int reopen_file(FILE * fileHandle)
+{
+	if(!fileHandle)
+		return 1;
+
+	int errcode = 0;
+
+	if(is_auto_log_name == 1)
+	{
+		errcode = sprint_auto_filename_format(&log_filename[0]);
+
+		if(errcode != 0)
+		{
+			running = 0;
+			return 1;
+		}
+	}
+
+	const char * fopen_opts = (sighup_count > 0) ? "a" : "w";
+	logfile = freopen(log_filename,fopen_opts,fileHandle);
+
+	running = (errcode == 0);
+	flag_reopen_file = 0;
+
+	return 0;
+}
+
+void get_logname_arg(const char * arg)
+{
+	if(arg != 0)
+	{
+		size_t len = strnlen(arg,MAXLOGNAMESZ);
```

Comment:
please add a newline after the variable declaration and add a space after the `,`